### ISSUE TITLE
CAPSImg for iOS

### ISSUE THREAD

**Issue by WeedyWeedSmoker** [lubomyr/uae4arm#33]
Sorry if this is the wrong place to post this, but can CAPSImg be compiled easily for iOS or does it need some rewriting?

Can you help me with this project? I'm maintaining the jailbroken iOS version of RetroArch and I would love to share a compatible version of CAPSImg, for iOS arm-64, and eventually for 32-bit too…

Thanks!

**Comment by lubomyr** [comment]
Hi, I think if it builded for arm64 for android devices, it can be also
buided for ios. As i remember i did not use any patch for arm64
compilation. I also tried to build uae4arm with this lib for macos with
apple m1 but stuck with compilation problem with another library mpeg2dec.
I did not found solution how to resolve it.


вт, 18 трав. 2021, 23:00 користувач Weedy Weed Smoker <
***@***.***> пише:

**Comment by WeedyWeedSmoker** [comment]
Now that you mention it, I think CAPSImg could work as-is on iOS arm64 if it compiled for m1…

Did you try to compile CAPSImg alone for m1 ?
Was the error with mpeg2dec separated, and related to uae4arm only ?


Thanks for the quick response!

**Comment by lubomyr** [comment]
I already do not remember if i tried to build CapsImg for m1. Error with
mpeg2dec not related with uae4arm, but this lib is required for built it
also as capsimg is required and some other libs.
If i will have some free time i will try to build it on next weekend.

вт, 18 трав. 2021, 23:45 користувач Weedy Weed Smoker <
***@***.***> пише:

**Comment by WeedyWeedSmoker** [comment]
@lubomyr Did you find any time to look into this?

Thanks!

**Comment by lubomyr** [comment]
@WeedyWeedSmoker 
Yes, I tried tobuild CapsImg for apple m1, and didn't notice any problem with compilation
just simple steps:
cd uae4arm/plugins/capsimage/CAPSImg
autoconf
./configure
make
<img width="563" alt="зображення" src="https://user-images.githubusercontent.com/468950/120098578-5a7a5880-c13f-11eb-8a9d-6c63c0f43e15.png">
all builded without any errors

**Comment by WeedyWeedSmoker** [comment]
@lubomyr That's great!

Do you know what would the command be to build a `capsimg.so` for iOS arm64 and armv7?

**Comment by lubomyr** [comment]
@WeedyWeedSmoker, I do not know. Sorry

**Comment by WeedyWeedSmoker** [comment]
@lubomyr Or maybe jut how to compile as another architecture, I just compiled on my Intel machine on macOS 11.4 and it only produces an x86_64 binary…

**Comment by WeedyWeedSmoker** [comment]
@lubomyr I think we can try with your compiled file… Could you send me the `libcapsimage.dylib` file to test it on iOS ?

**Comment by WeedyWeedSmoker** [comment]
@lubomyr I think I managed to do it… Let me try first!

**Comment by adespoton** [comment]
Think you could put that capsimg.so up on your page? I'm still not finding a prebuilt anywhere and don't want to re-create the journey you took in getting it to compile....

**Comment by JohanMargueritte** [comment]
@adespoton capsimage.so for iOS 32 & 64-bit : https://www.mediafire.com/file/dafvb1roszmsmbr/capsimg.so/file

**Comment by adespoton** [comment]
Thanks; oddly, RetroArch now crashes without warning instead of complaining about a lack of capsimg.so.

-Em
